src/internal/common/jwt/jwt.go

```diff
@@ -0,0 +1,39 @@
+package jwt
+
+import (
+	"time"
+
+	"github.com/alcionai/clues"
+	jwt "github.com/golang-jwt/jwt/v5"
+)
+
+// IsJWTExpired checks if the JWT token is past expiry by analyzing the
+// "exp" claim present in the token. Token is considered expired if "exp"
+// claim < current time. Missing "exp" claim is considered as non-expired.
+// An error is returned if the supplied token is malformed.
+func IsJWTExpired(
+	rawToken string,
+) (bool, error) {
+	p := jwt.NewParser()
+
+	// Note: Call to ParseUnverified is intentional since token verification is
+	// not our objective. We only care about the embed claims in the token.
+	// We assume the token signature is valid & verified by caller stack.
+	token, _, err := p.ParseUnverified(rawToken, &jwt.RegisteredClaims{})
+	if err != nil {
+		return false, clues.Wrap(err, "invalid jwt")
+	}
+
+	t, err := token.Claims.GetExpirationTime()
+	if err != nil {
+		return false, clues.Wrap(err, "getting token expiry time")
+	}
+
+	if t == nil {
+		return false, nil
+	}
+
+	expired := t.Time.Unix() < time.Now().Unix()
```

Comment:
>JWT producer may be in a different timezone compared to consumer. Although this can be addressed by converting both to UTC.

Wouldn't timezone be an issue in either case? It's just not apparent when converting to Unix since timezone info is elided during the conversion. `After` and `Before` also [take into account timezones](https://play.golang.com/p/mCT6W3z_8Fi)

It seems like we should probably figure out what timezone the parsed value from the JWT token is created in. `time.Now()` says it uses the local timezone

Reply:
https://chat.openai.com/share/9550ff9a-5077-459d-a35e-47c20890240e

Reply:
> Wouldn't timezone be an issue in either case? It's just not apparent when converting to Unix since timezone info is elided during the conversion. It seems like we should probably figure out what timezone the parsed value from the JWT token is created in

Timezone is not an issue because unix timestamps are [timezone independent](https://stackoverflow.com/a/23062640/2483500). Machine A in NY and Machine B in WA would produce the same unix timestamp at a given point in time. 

> After and Before also take into account timezones

Thanks! I'll switch to this, although I don't see any issues with existing code.

Reply:
@vkamra My first commit in this PR was based off an older version of `golang-jwt/jwt` which is a clone of `dgrijalva/jwt-go`, same package used by chatgpt. It has a [verifyExpiresAt](https://github.com/dgrijalva/jwt-go/blob/9742bd7fca1c67ba2eb793750f56ee3094d1b04f/map_claims.go#L36) func which does time comparisons after converting to `Unix()`, like I am doing right now. Here is my old code [code](
https://github.com/alcionai/corso/pull/4417/commits/f03eeefd903d287c7189adab4feed43331aa3498#diff-74f564165519040be446f0254397adc4f6e516f15145d1de5b286a4bd564eb3cR26) which uses `verifyExpiresAt`. 

This func was deprecated(unexposed) in `golang-jwt/jwt/v5` as they put things behind a validator which is unexported & still in flux. 

We can use either unix timestamp comparisons or time.before/after like Ashlie suggested. I think both approaches are fine and will work.

Reply:
I'll add some tests with jwt producer/consumer in different timezones.

Reply:
> @vkamra My first commit in this PR was based off `dgrijalva/jwt-go`, same package used by chatgpt. It has a [verifyExpiresAt](https://github.com/dgrijalva/jwt-go/blob/9742bd7fca1c67ba2eb793750f56ee3094d1b04f/map_claims.go#L36) func which does time comparisons after converting to `Unix()`, like I am doing right now. Here is my old [code](
> 
> https://github.com/alcionai/corso/pull/4417/commits/f03eeefd903d287c7189adab4feed43331aa3498#diff-74f564165519040be446f0254397adc4f6e516f15145d1de5b286a4bd564eb3cR26) which uses `verifyExpiresAt`. 
> 
> 
> 
> This func was deprecated(unexposed) in `golang-jwt/jwt/v5`.  They put it behind a validator which is currently not accessible for `ParseUnverified`. 
> 
> 
> 
> We can use either unix timestamp comparisons or time.before/after like Ashlie suggested. I think both approaches are fine and will work. 
> 
> 
> 
> 
> 
> 

Yeah sorry I should have provided context. I pasted that not to suggest we should switch the library but to provide input on the time zone question and also that we should use time.After/Before. Agree with your proposal.